### PR TITLE
[#10920] Add front-end unit tests for FeedbackResponsesService

### PR DESCRIPTION
**Pull request by jayasting98**
Part of #10920 

**Outline of Solution**

Front-end unit tests were added for each method in `FeedbackResponsesService`.

### CODE REVIEW

**Comment by jianhandev** [comment]
@AdithyaNarayan appreciate your review on this PR!

File: src/web/services/feedback-responses.service.spec.ts [hunk 1]
```diff
@@ -1,16 +1,531 @@
...
+
+  it('should retrieve the correct default distribute points/constant sum (options) response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.CONSTSUM_OPTIONS))
+      .toStrictEqual(DEFAULT_CONSTSUM_RESPONSE_DETAILS());
+  });
+
+  it('should retrieve the correct default distribute points/constant sum (recipients) response details', () => {
+    expect(service.getDefaultFeedbackResponseDetails(FeedbackQuestionType.CONSTSUM_RECIPIENTS))
+      .toStrictEqual(DEFAULT_CONSTSUM_RESPONSE_DETAILS());
+  });
```
Comment (AdithyaNarayan) [inline src/web/services/feedback-responses.service.spec.ts:116]: Only the parameters seem to be different across all these test cases. Why not use a `Record`/List to store pairs of `QuestionType` and Response details and then loop through them? Or add them in the same test case?

File: src/web/services/feedback-responses.service.spec.ts [hunk 2]
```diff
@@ -1,16 +1,531 @@
...
+    feedbackResponseDetails.answers = [null as unknown as number, 1, RANK_OPTIONS_ANSWER_NOT_SUBMITTED];
+    expect(service.isFeedbackResponseDetailsEmpty(feedbackQuestionType, feedbackResponseDetails))
+      .toBeFalsy();
+
+    feedbackResponseDetails.answers = [1];
+    expect(service.isFeedbackResponseDetailsEmpty(feedbackQuestionType, feedbackResponseDetails))
+      .toBeFalsy();
+
+    feedbackResponseDetails.answers = [2, 1];
+    expect(service.isFeedbackResponseDetailsEmpty(feedbackQuestionType, feedbackResponseDetails))
```
Comment (AdithyaNarayan) [inline src/web/services/feedback-responses.service.spec.ts:200]: A lot of duplicate code here too

**Review by AdithyaNarayan — CHANGES_REQUESTED** [review]

**Comment by ypinhsuan** [comment]
Hi @jayasting98, any updates on this PR?

**Comment by jayasting98** [comment]
@ypinhsuan Sorry, I have been a bit busy. Let me see if I can try to do it next week.

**Comment by jayasting98** [comment]
@AdithyaNarayan @ypinhsuan Sorry it took a while; I was busy. I'm done now.

**Review by AdithyaNarayan — APPROVED** [review]
LGTM

**Review by ypinhsuan — APPROVED** [review]
Looks good!

File: src/web/services/feedback-responses.service.spec.ts [hunk 3]
```diff
@@ -1,16 +1,429 @@
...
+    const section: string = 'giver section';
+    const sectionType: InstructorSessionResultSectionType = 'UNKNOWN' as InstructorSessionResultSectionType;
+    expect(service.isFeedbackResponsesDisplayedOnSection(response, section, sectionType))
+      .toBeTruthy();
+  });
+
+  it('should call get when retrieving a feedback response', () => {
+    const dummyIntent: Intent = Intent.STUDENT_SUBMISSION;
+    const paramMap: Record<string, string> = {
+      questionid: '[auto-generated Datastore ID]',
```
Comment (wkurniawan07) [inline src/web/services/feedback-responses.service.spec.ts:388]: Don't make any mention to "Datastore" or anything DB-related here. Even 'dummy question id' is sufficient.

File: src/web/services/feedback-responses.service.spec.ts [hunk 4]
```diff
@@ -1,16 +1,429 @@
...
+      intent: dummyIntent,
+      key: '[generated registration key]',
```
Comment (wkurniawan07) [inline src/web/services/feedback-responses.service.spec.ts:390]: No need to mention "generated" either. Front-end doesn't care how the values are created at the back-end.

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**